### ISSUE TITLE
Sublime crashes after saving new empty note on tab close

### ISSUE THREAD

**Issue by pimlock** [bordaigorl/sublime-evernote#93]
Steps to reproduce:
1. Make sure that "warn_on_save" setting is set to "true"
2. Create new empty note ("Evernote: New empty note")
3. Set notebook, title, add contents
4. Close the tab
5. Choose "Save to Evernote and close"

Note is saved, but the whole sublime crashes.

**Comment by bordaigorl** [comment]
Hi! Thanks for making the effort of fixing this!
I cannot seem to be able to reproduce the issue.
Could you specify which system/ST version are you working on?

**Comment by pimlock** [comment]
Hi! Thanks for looking into that! 

I totally forgot to provide versions. Sorry about that.
* Sublime Text 3 (build 3083)
* sublime-evernote (I'm using devel branch - because it had code that asks for confirmation when closing the tab with new note)
* Mac OS 10.10.3

Little bit background for this change - I created new empty note and after I wrote for 15 minutes I mistakenly closed the tab and it was gone, as it didn't ask me whether I want to save it or discard (I have 'warn_on_close' on).
So I thought that I will spend some time and add this as it should be simple feature. So I updated on_pre_close method and added this 'on_completion' callback as getting note is async. Then when I was about to create pull request I noticed that similar fix was already in devel branch, so I tried it. But whenever I tried to create new note, close the tab and choose 'Save' it would crash my Sublime. I thought that this may be caused by the fact that tab is closed before note is saved (as saving is async and tab is closed just after calling async method, which will (in most cases) happen before note is actually saved. So I took this part of my code and created this pull request.

Anyway, it's more like a suggestion for adding callbacks to async method as I believe it's cleaner way to handle this kind of behaviour.
